javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -0,0 +1,1024 @@
+        public BitSet<T> take(long n) {
+            if (n <= 0) {
+                return createEmpty();
+            } else if (n >= length()) {
+                return this;
+            } else {
+                return createFromAll(iterator().take(n));
+            }
+        }
+
+        @Override
+        public BitSet<T> takeRight(long n) {
+            if (n <= 0) {
+                return createEmpty();
+            } else if (n >= length()) {
+                return this;
+            } else {
+                return createFromAll(iterator().takeRight(n));
+            }
+        }
+
+        @Override
+        public BitSet<T> remove(T t) {
+            if (contains(t)) {
+                final int element = toInt.apply(t);
+                final long[] copy = copyExpand(getWordsNum());
+                unsetElement(copy, element);
+                return fromBitMaskNoCopy(shrink(copy));
+            } else {
+                return this;
+            }
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public BitSet<T> removeAll(Iterable<? extends T> elements) {
+            final Stream<Integer> source = Stream.ofAll(elements).map(toInt);
+            final long[] copy = copyExpand(getWordsNum());
+            source.forEach(element -> {
+                unsetElement(copy, element);
+            });
+            return fromBitMaskNoCopy(shrink(copy));
+        }
+
+        @Override
+        public String toString() {
+            return mkString(stringPrefix() + "(", ", ", ")");
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            } else if (o instanceof BitSet) {
+                final BitSet<?> that = (BitSet<?>) o;
+                return Collections.equals(this, that);
+            } else {
+                return false;
+            }
+        }
+
+        @Override
+        public int hashCode() {
+            return Collections.hash(this);
+        }
+    }
+
+    class BitSet1<T> extends AbstractBitSet<T> {
+
+        private static final long serialVersionUID = 1L;
+
+        private final long elements;
+
+        BitSet1(Function1<Integer, T> fromInt, Function1<T, Integer> toInt, long elements) {
+            super(fromInt, toInt);
+            this.elements = elements;
+        }
+
+        @Override
+        int getWordsNum() {
+            return 1;
+        }
+
+        @Override
+        long[] copyExpand(int wordsNum) {
+            if(wordsNum < 1) {
+                wordsNum = 1;
+            }
+            long[] arr = new long[wordsNum];
+            arr[0] = elements;
+            return arr;
+        }
+
+        @Override
+        long getWord(int index) {
+            if (index == 0) {
+                return elements;
+            } else {
+                return 0L;
+            }
+        }
+
+        @Override
+        public T head() {
+            if (elements != 0) {
```

Comment:
could you please invert the if, usually the guard is in the first branch throughout the codebase :)